server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -89,7 +89,7 @@ public class BalancedShardsAllocator implements ShardsAllocator {
     public static final Setting<Float> THRESHOLD_SETTING = Setting.floatSetting(
         "cluster.routing.allocation.balance.threshold",
         1.0f,
-        0.0f,
+        1.0f,
```

Comment:
Unfortunately tightening the bound is a breaking change so I don't think we can do it like this. Instead we should continue to accept values between 0 and 1, but replace such values with 1 (and emit some deprecation warning).